### PR TITLE
dispatcher: Refuse to start extra dispatchers

### PR DESCRIPTION
**Pull request by zmc**

### CODE REVIEW

**Comment by zmc** [comment]
@dmick CI is failing because the github actions don't support Python 3.6 anymore. We should talk about upgrading to 3.8 on the teuthology VM.

File: teuthology/dispatcher/__init__.py [hunk 1]
```diff
@@ -184,6 +194,29 @@ def main(args):
...
+    user = getpass.getuser()
+    def match(proc):
+        if proc.username() != user:
+            return False
+        cmdline = proc.cmdline()
+        if len(cmdline) < 2:
+            return False
+        if not cmdline[1].endswith("/teuthology-dispatcher"):
+            return False
+        if cmdline[2] == "--supervisor":
```
Comment (jdurgin) [inline teuthology/dispatcher/__init__.py:207]: Check for 3 args before access?

**Review by jdurgin — APPROVED** [review]

**Review by dmick — COMMENTED** [review]
LGTM.  I spent some time trying to discover what exactly starts teuthology-dispatcher without much success, because I was wondering how it might come to pass that something would try to start two..maybe we could add a little theory-of-operation to the doc?  (or am I missing it)

**Review by dmick — APPROVED** [review]

**Comment by zmc** [comment]
#1798 aims to get the CI working again

**Comment by dmick** [comment]
Holding off on this until @zmc can get to the bottom of why the checks are failing for #1798

**Review by dmick — APPROVED** [review]
LGTM, green tests, go

**Pull request merged**